### ISSUE TITLE
LibreOffice Writer and LanguageTool extension - doesn't work (Linux)

### ISSUE THREAD

**Issue by kacnje** [languagetool-org/languagetool#6525]
I have 64-bit Linux Mint 20.3 with LibreOffice 6.4.7.2. I installed LanguageTool Extension (latest version 5.7). 
In LibreWriter I have menu Tools -> LanguageTool (with 7 items) and five icons on the toolbar. 
But clicking on menu items or clicking on icons doesn't do anything. 
What can be wrong? 
The latest [reviews here ](https://extensions.libreoffice.org/en/extensions/show/languagetool) are quite bad. So I am not alone with this issue.

**Comment by danielnaber** [comment]
Have you tried the tips at https://languagetooler.freshdesk.com/support/solutions/folders/44001214393?

**Comment by FredKruse** [comment]
@kacnje Could you fix your problem?
Most implementation problems on Linux systems arise because the package libreoffice-java-common isn't installed by default. Run "sudo apt-get install libreoffice-java-common" inside a terminal, to add the support of the java interface of LibreOffice.

**Comment by kacnje** [comment]
Sorry for delay ... the package libreoffice-java-common was installed. 

@danielnaber: I checked mentioned tips before I wrote a question here. 

Is maybe a log file somewhere on computer where I can check what is going on?

**Comment by FredKruse** [comment]
This is very strange. Linux Mint is my favorite system. I'm using it on different virtual machines and on a separate notebook. There wasn't any problems. So, I think we should do a systematic analyses.
@kacnje please do the following tests and answer the questions:

1. Open an empty document. Click on the LanguageTool (LT) Option Button (blue wavy underlined gear wheel). Does the LT option panel appear? (this can take 1 to 3 seconds)
2. Type a short sentence in your default language. Type two blanks between two words. Appears a green wavy line under the two blanks? Which language do you use?
3. If the first two points work, every thing is fine, and we can stop here. When not: Look at ~/.config/LanguageTool/LibreOffice/. There should be a file LanguageTool.log and possibly a file LanguageTool.cfg. Please post them here.
4. In LibreOffice, open the menu point tools/options. In the option window, go to libreoffice/advanced. The checkbox "Use a Java runtime environment" should be enabled. Which version of JRE is selected?

**Comment by kacnje** [comment]
1. No. It doesn't appear. 
2. I wrote some words with two blanks between the words. There was nothing underlined. I use the Slovenian language. 
3. 
LOG file:
LT office integration log from Tue Jun 14 10:00:01 CEST 2022
Java-Version: 11.0.15, max. Heap-Space: 3968 MB, LT Heap Space Limit: 3571 MB
Configured Linguistic Service: org.languagetool.openoffice.Main, sl-SI
Test for document language failed: Can't trigger event: configure

CFG file **(!!! the URL is changed here !!!)**:
motherTongue=sl-SI
ltVersion=5.7
#Profile: Default
#Tue Apr 12 11:33:11 CEST 2022
serverMode=false
taggerShowsDisambigLog=false
useGUIConfig=false
autoDetect=false
useOtherServer=true
useDocumentLanguage=false
fixedLanguage=sl-SI
numberParagraphs=-2
serverPort=8081
otherServerUrl=https\://one.two.si/XXXXXXX/api
noDefaultCheck=true

URL used is valid and has our own LT server for Slovenian. It is used by many users and it works without problem except here in LibreOffice. Maybe just a question - is a syntax for URL valid? I can't access to Options dialog and now I saw that maybe the problem is in syntax?

4. The checkbox is checked. Version is 11.0.15.

**Comment by FredKruse** [comment]
I think I got the problem:
It seems the automatic grammar check is disabled. Please enable "Check grammar as you type" at "Tools"/"Options"/"Language Settings"/"Writing Aids" and restart LibreOffice/OpenOffice.
If this solves the Problem, close the issue, please.
I added a warning in version 5.8.

**Comment by kacnje** [comment]
@FredKruse - I checked the settings and found "Check grammar as you type" was enabled. But the setting for spelling was disabled and enabling it, does the job. 
The Options appeared once but after that, I got "Connection refused: Switch to internal LanguageTool application". I checked in LOG file and I found:
_java.lang.NullPointerException
Server doesn't support maxTextLength, Limi text length set to: 20000._ 

I will send this to our "LanguageTool server" developers to check.
